### PR TITLE
Destination Snowflake: cleaning up unused code paths

### PR DESCRIPTION
**Pull request by gisripa**
## What
Remove remnants of `SwitchingDestination` and decouple Snowflake from `AbstractJdbcDestination` base implementation (nothing is reused except a check related function)

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by gisripa** [comment]
* **#38553** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38553?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java [hunk 1]
```diff
@@ -69,9 +69,9 @@ public String createTableQuery(final JdbcDatabase database, final String schemaN
 
   /**
    * Sort of hacky. The problem is that SnowflakeSqlOperations is constructed in the
-   * SnowflakeDestination constructor, but we don't have the JsonNode config until we try to call
-   * check/getSerializedConsumer on the SnowflakeDestination. So we can't actually inject the config
-   * normally. Instead, we just use the singleton object. :(
+   * SnowflakeDestinationOld constructor, but we don't have the JsonNode config until we try to call
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java:72]: in between refactor, intellij used the tmp class name. will rename it in comment.

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/JdbcCheckOperations.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                namingResolver.getIdentifier(
+                    "_airbyte_connection_test_" +
+                        UUID.randomUUID().toString().replace("-".toRegex(), ""),
+                )
+            sqlOps.createSchemaIfNotExists(database, outputSchema)
+            sqlOps.createTableIfNotExists(database, outputSchema, outputTableName)
+            // verify if user has permission to make SQL INSERT queries
+            try {
+                if (attemptInsert) {
+                    sqlOps.insertRecords(
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/JdbcCheckOperations.kt:73]: sigh. snowflake shouldn't even be using this method. fine for now, but eventually we should have snowflake do the actual internal stage + COPY thing for `check`

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java (hunk 1 above)
Comment (edgao) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java:72]: what is SnowflakeDestinationOld ?

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java [hunk 3]
```diff
@@ -361,4 +362,9 @@ private String toJdbcTypeName(final AirbyteProtocolType airbyteProtocolType) {
     };
   }
 
+  @Override
+  public void createNamespaces(@NotNull Set<String> schemas) {
+
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java:367]: nit: is it safe to `throw new UnsupportedOperationException("not yet implemented")`? (just to be explicit that this _should_ be implemented, we just haven't gotten around to it yet

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.kt [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+) : BaseConnector(), Destination {
+
+    override fun check(config: JsonNode): AirbyteConnectionStatus? {
+        val snowflakeInternalStagingSqlOperations =
+            SnowflakeInternalStagingSqlOperations(nameTransformer)
+        val dataSource = getDataSource(config)
+        try {
+            val database = getDatabase(dataSource)
+            val outputSchema = nameTransformer.getIdentifier(config["schema"].asText())
+            attemptTableOperations(
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.kt:62]: mega nit: I'd prefer this to not be statically imported, so it's easier to see that it's defined in the Jdbc class

File: airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java [hunk 5]
```diff
@@ -74,18 +73,6 @@ void testUrlPattern(final String url, final boolean isMatch) throws Exception {
     assertEquals(isMatch, matcher.find());
   }
 
-  @ParameterizedTest
-  @MethodSource("destinationTypeToConfig")
-  public void testS3ConfigType(final String configFileName, final DestinationType expectedDestinationType) throws Exception {
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java:79]: doublechecking: this test was important for when we supported s3/gcs staging, but we deleted those so now we can delete the test?

**Review by edgao — APPROVED** [review]
tiny nits. I only skimmed SnowflakeDestination, since I assume that's mostly just autoconverted code?

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java (hunk 1 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java:72]: see my comment above. in-between refactor that was the tmp class 😅

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java (hunk 3 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java:367]: yeah was planning to add that. just to be sure it doesn't get called erroneously.

File: airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java (hunk 5 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java:79]: Yeah. something something SwitchingDestination.. destinationType check. which is always InternalStaging

**Pull request merged**